vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1208,11 +1208,14 @@ func resourceVSphereVirtualMachineCreateBare(d *schema.ResourceData, meta interf
 	// environment info in the resource pool, which we can then filter through
 	// our device CRUD lifecycles to get a full deviceChange attribute for our
 	// configspec.
-	var guestID interface{}
-	if _, ok := d.GetOk("guest_id"); !ok {
+	var guestID string
```

Comment:
Rather than having an else block to set the default case, what about this:

```
guestID := "other-64"
if guestInterface, ok := d.GetOk("guest_id"); ok {
    guestID = guestInterface.(string)
} 
```

Reply:
Thank you!!! For whatever reason, I could not see that yesterday!